src/main/java/org/testng/internal/ClassBasedParallelWorker.java

```diff
@@ -49,7 +49,7 @@
         params = getParameters(im);
         prevClass = c;
       }
-      if (sequentialClasses.contains(c)) {
+      if (sequentialClasses.contains(c) || isSame(sequentialClasses, c)) {
```

Comment:
I think `sequentialClasses.contains(c)` is useless.
If not, prefer to add it into the new method

Reply:
`sequentialClasses.contains(c)` is needed. I moved it into the new method.